### PR TITLE
Tutorial: Custom keybinding in PTK

### PR DESCRIPTION
**Pull request by gforsyth**
I could probably add a little more on filters and a few more examples, but this is ready for a glance.

### CODE REVIEW

**Comment by astronouth7303** [comment]
Could we get some special warning/note boxes?

File: docs/tutorial_ptk.rst [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+All of the code below can be entered into your `xonshrc <xonshrc.html>`_
+
+Control characters
+==================
+
+We can't and won't stop you from doing what you want, but in the interest of a
+functioning shell, you probably shouldn't mess with the following keystrokes.
+Some of them are `ASCII control characters
+<https://en.wikipedia.org/wiki/Control_character#In_ASCII>`_ and _really_
```
Comment (astronouth7303) [inline docs/tutorial_ptk.rst]: GitHub isn't formatting the `_really_`.

File: docs/tutorial_ptk.rst [hunk 2]
```diff
@@ -0,0 +1,139 @@
+.. _tutorial_ptk:
+
+***********************************************
+Tutorial: ``prompt_toolkit`` custom keybindings
+***********************************************
+
+Are you really jonesing for some special keybindings? We can help you out with
+that. The first time is free and so is every other time!
+
+.. warning:: This tutorial will let you hook directly into the
```
Comment (astronouth7303) [inline docs/tutorial_ptk.rst]: This appears to be producing an unadorned `<div>`, which will be hard to style. GitHub quirk or how RST works?

File: docs/tutorial_ptk.rst [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+      - SIGINT current job
+
+
+Useful imports
+==============
+
+There are a few useful ``prompt_toolkit`` tools that will help us create better
+bindings::
+
+    from prompt_toolkit.keys import Keys
```
Comment (astronouth7303) [inline docs/tutorial_ptk.rst]: Does the Python syntax highlighter work well enough for this?

File: docs/tutorial_ptk.rst (hunk 2 above)
Comment (scopatz) [inline docs/tutorial_ptk.rst]: This is valid rst

File: docs/tutorial_ptk.rst (hunk 1 above)
Comment (scopatz) [inline docs/tutorial_ptk.rst]: Githubs rst parser is woefully insufficient and probably not something we should really worry about here.

File: docs/tutorial_ptk.rst (hunk 3 above)
Comment (scopatz) [inline docs/tutorial_ptk.rst]: Yes, there is the double colon above, so this will be interreted as a code block, which defaults to Python

**Comment by scopatz** [comment]
Since this is a doc-only change I am going to go ahead and merge. Thanks so much for writing this @gforsyth!

**Pull request merged**